File: Data Structures & Algorithms/is-palindrome/submission-0.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        int i = 0;
        int j = s.length() - 1;

        boolean ok = true;
        while(i < s.length() && j > 0){
            if(!Character.isLetterOrDigit(s.charAt(i))){
                i++;
                continue;
            }
            if(!Character.isLetterOrDigit(s.charAt(j))){
                j--;
                continue;
            }
            if(s.charAt(i) != s.charAt(j) ){
                ok = false;
                break;
            }

            i++;
            j--; 
        }
        return ok;
    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.cpp
class Solution {
public:

    bool isPalindrome(string s){
        int i = 0;
        int n = s.size();
        while(i < n / 2){
            if(s[i] != s[n-i-1])
                return false; 
            i++;
        }
        return true;
    }

    bool validPalindrome(string s) {
        int i = 0;
        if(isPalindrome(s)) return true;
        while(i < s.size()){
                cout << s.substr(0, i) + s.substr(i+1) << endl;
            if(isPalindrome(s.substr(0, i) + s.substr(i+1))){
                return true;
            }
            i++;
        }
        return false;
    }
};